plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/FordiacMarkerResolutionGenerator.java

```diff
@@ -30,9 +30,10 @@ public IMarkerResolution[] getResolutions(final IMarker marker) {
 		case LibraryElementValidator.ITYPED_ELEMENT__VALIDATE_TYPE,
 				LibraryElementValidator.CONFIGURABLE_FB__VALIDATE_DATA_TYPE ->
 			new IMarkerResolution[] { new ChangeDataTypeMarkerResolution(marker),
-					new CreateDataTypeMarkerResolution(marker) };
-		case LibraryElementValidator.TYPED_CONFIGUREABLE_OBJECT__VALIDATE_TYPE -> new IMarkerResolution[] {
-				new CreateMissingFBMarkerResolution(marker), new ChangeFBMarkerResolution(marker) };
+					new CreateDataTypeMarkerResolution(marker), new BestFitDataTypeMarkerResolution(marker) };
+		case LibraryElementValidator.TYPED_CONFIGUREABLE_OBJECT__VALIDATE_TYPE ->
+			new IMarkerResolution[] { new CreateMissingFBMarkerResolution(marker), new ChangeFBMarkerResolution(marker),
+					new BestFitFBMarkerResolution(marker) };
```

Comment:
I believe the intention was to add a separate resolution for each result of `findUnqualified(...)` that shows the full-qualified name so users can choose potential matches quickly, without going through an additional dialog. The resolution would then simply need to apply its respective type.

Reply:
quick fixes based on names now show up in the initial quick fix dialog